### DISCUSSION TITLE
Easily receive specific item revision from API

### DISCUSSION THREAD

**Discussion by PaulRaschke** [directus/directus#6603] [Feature Requests]
Currently, retrieving a specific revision of an item is complicated, especially with a GraphQL query, because only an (untyped) JSON snapshot is stored in the revision item in the DB. Manually parsing the JSON snapshot, resolving relational fields, filtering revisions based on field values, etc. would involve a lot of additional logic outside of the query definition and probably requires fetching many more revisions than necessary.

It would be great to be able to easily query revisions (via GraphQL) and also be able to select specific item revisions based on a filter definition through a possibly additional query parameter.
Perhaps this could be accomplished by adding a virtual revision field to the item?

My current use-case:
I want to retrieve the latest version of an item where the status was "published".
This use case might also be related to https://github.com/directus/directus/discussions/2975.

Directus version used:
v9.0.0-rc.83 (PostgreSQL 12)

**Comment by fg-uulm** [comment]
Chiming in on this - much needed! Is there a simple way to make this work?

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
